Invalid .whl name compiling inside a python virtual (venv) environment

Hi
    I had previously tried to compile Pangolin inside a system environment, and, as I was not using a venv, (as is now required), I chose to force the pip install using "--break-system-packages". This compiled and installed fine, and I could run the examples. Unfortunately, as may have been expected, I broke my system packages. and thus had to remove the Pangolin elements.

  I have now tried to compile Pangolin inside a venv, and the wheel is not being correctly created. (This is the latest Pangolin from github with the Python_EXECUTABLE swapped for Python3_EXECUTABLE changes)

This is on a raspberry pi4 running Bookworm. In order to keep things simple: This a 100% fresh sdcard install with no other software on the card with all  OS updates included.


  These were my commands:

```sh
git clone --recursive https://github.com/stevenlovegrove/Pangolin.git
 cd Pangolin
./scripts/install_prerequisites.sh recommended
cmake -B build
cmake --build build -t pypangolin_pip_install
```

This runs without any errors until the final python .whl install (The .whl has an empty name) Output is as below

```
[ 98%] Linking CXX shared module pypangolin-0.9.2.data/purelib/pypangolin.cpython-311-aarch64-linux-gnu.so
lto-wrapper: warning: using serial compilation of 23 LTRANS jobs
lto-wrapper: note: see the ‘-flto’ option documentation for more information

[ 98%] Built target pypangolin
[ 98%] Creating Wheel /home/robot/LearnOpenCV/Pangolin/build/.whl
[ 98%] Built target pypangolin_wheel
[100%] Installing for selected Python '/home/robot/LearnOpenCV/bin/python3'
ERROR: Invalid requirement: '/home/robot/LearnOpenCV/Pangolin/build/.whl'
Hint: It looks like a path. The path does exist. 
gmake[3]: *** [CMakeFiles/pypangolin_pip_install.dir/build.make:71: CMakeFiles/pypangolin_pip_install] Error 1
gmake[2]: *** [CMakeFiles/Makefile2:957: CMakeFiles/pypangolin_pip_install.dir/all] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:964: CMakeFiles/pypangolin_pip_install.dir/rule] Error 2
gmake: *** [Makefile:390: pypangolin_pip_install] Error 2
```

I am assuming that the switch to a venv has caused this, but I am unsure

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -89,7 +89,7 @@ jobs:
     - name: "(vcpkg only) install setuptools"
       if: ${{ matrix.package_manager == 'vcpkg' }}
       run: |
-        pip install setuptools
+        pip install setuptools wheel
 
     - name: Configure CMake
       run: >

--- a/cmake/MakePythonWheel.cmake
+++ b/cmake/MakePythonWheel.cmake
@@ -46,8 +46,12 @@ print(wheel_name(name='${python_module}', version='${version}', ext_modules=[Ext
 "
         OUTPUT_VARIABLE wheel_filename
         OUTPUT_STRIP_TRAILING_WHITESPACE
-        ERROR_QUIET
+        RESULT_VARIABLE has_wheel_filename
     )
+
+    if(NOT has_wheel_filename EQUAL "0")
+        message(FATAL_ERROR "Cannot format wheel filename via 'setuptools'.")
+    endif()
 
     set(wheel_filename "${CMAKE_BINARY_DIR}/${wheel_filename}.whl")
     set(wheel_distinfo "${CMAKE_BINARY_DIR}/${python_module}-${version}.dist-info")

--- a/cmake/MakePythonWheel.cmake
+++ b/cmake/MakePythonWheel.cmake
@@ -28,6 +28,23 @@ except ImportError as e:
 
     execute_process(
         COMMAND ${Python3_EXECUTABLE} -c "
+import sys
+try:
+    from wheel.bdist_wheel import bdist_wheel
+    sys.exit(0)
+except ImportError as e:
+    print(f'{e}. Search paths:', file=sys.stderr)
+    for p in sys.path: print(f'  {p}', file=sys.stderr)
+    sys.exit(1)
+"
+    RESULT_VARIABLE has_bdist_wheel)
+
+    if(has_bdist_wheel EQUAL "1")
+        message(FATAL_ERROR "Python module `wheel.bdist_wheel` required for correct wheel filename generation.")
+    endif()
+
+    execute_process(
+        COMMAND ${Python3_EXECUTABLE} -c "
 from setuptools.dist import Distribution
 from setuptools import Extension
 

